### PR TITLE
feat:user profile backend

### PR DESCRIPTION
**Pull request by VitoKondor**

### CODE REVIEW

File: backend/recipeGraphQLSchema.ts [hunk 1]
```diff
@@ -5,6 +5,7 @@ export const buildRecipeSchema = () => {
     fields: (t) => ({
       id: t.exposeID('id'),
       name: t.exposeString('name'),
+      userId: t.exposeString('userId', { nullable: true }),
```
Comment (Bhoft) [inline backend/recipeGraphQLSchema.ts:8]: why nullable?

File: prisma/schema.prisma [hunk 2]
```diff
@@ -21,6 +21,8 @@ model Recipe {
   recipeIngredients RecipeIngredient[]
   recipeEvaluations RecipeEvaluation[]
   recipeTags RecipeTags[]
+  userId String?
```
Comment (Bhoft) [inline prisma/schema.prisma:24]: maybe better profileId / userProfileId instead of userId

File: prisma/schema.prisma [hunk 3]
```diff
@@ -67,3 +69,15 @@ model Tag {
    color String?
    recipeTags RecipeTags[]
 }
+
+model UserProfile {
+  id    String  @id @default(cuid())
+  userName  String
+  country String?
+  city String?
+  countryRank String?
```
Comment (Bhoft) [inline prisma/schema.prisma:78]: what is countryRank ?

File: pages/userEdit.page.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+import { useQuery } from 'urql'
+import { Layout } from '../components/layout/layout'
+import UserForm from '../components/userDetails/userEdit'
+
+const Details = () => {
+  const session = useSession()
+
+  const [result] = useQuery({
+    query: `
+            query userByName($userName:String!) {
```
Comment (Bhoft) [inline pages/userEdit.page.tsx:12]: userByName => userDetailByName

File: prisma/seed.ts [hunk 5]
```diff
@@ -3,6 +3,33 @@ import { recipes, ingredients, tags } from './seed.data'
 
 const prisma = new PrismaClient()
 async function main() {
+  await prisma.userProfile.upsert({
```
Comment (Bhoft) [inline prisma/seed.ts:6]: move to seed.data.ts

**Review by Bhoft — APPROVED** [review]

**Review by clemens1337 — APPROVED** [review]

**Pull request merged**